### ISSUE TITLE
Broken environment definition file

### ISSUE THREAD

**Issue by Molkree** [tardis-sn/tardis#1564]
**Describe the bug**
In #1549 `compilers` was [removed](https://github.com/tardis-sn/tardis/commit/923e59dda979758378c07f9cd9884e91681149a0#diff-58b2a816009fd40ceaed5b3cac419bb7b2b24491e0128141b9e3bb5a69e3adacL15-L16) from the env file.
As a result conda environment build fails on the pip stage: `CondaEnvException: Pip failed`. This is because `pickle5` can't be built without gcc.
```
  unable to execute 'gcc': No such file or directory
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pickle5
```

[Full console log](https://github.com/tardis-sn/tardis/files/6467084/error_log.txt).

**To Reproduce**
Just follow tardis installation steps up to `conda env create -f tardis_env3.yml` step.

**Screenshots**
![image](https://user-images.githubusercontent.com/12860284/117998861-f6891f00-b34c-11eb-9eb3-ce7321098949.png)
...
![image](https://user-images.githubusercontent.com/12860284/117998932-07399500-b34d-11eb-949d-ea67e6d0a870.png)

**System**
 - OS:
    - [x] GNU/Linux
    - [ ] macOS

**Additional context**
You can actually activate tardis environment even after this conda exception. But of course it will lack pip dependencies (though they are only used to build docs according to comment in the file?) and I'm not sure if conda does any steps after pip which could cause something else breaking.

I'm also using Python 3.8, so pickle5 is not even needed as it's a port of Python 3.8.3 pickle to Python 3.5-3.7. Maybe `pickle5 ; python_version < '3.8'` will work. We also don't need to redefine pickle on all Python versions, only < 3.8:
https://github.com/tardis-sn/tardis/blob/e3a35bb09be710e1f63ffc586c28c61f2857da05/tardis/__init__.py#L37-L39

P.S. ~~Maybe add CI check to try and build environment (and install tardis in it?) when env file (and setup.py?) change.~~ It's in the `documentation-build` workflow.

**Comment by epassaro** [comment]
I think this is not failing for us (or at the pipelines) because almost every Linux distro has `gcc` installed by default (or at some point) through the package manager.

I would try to move `pickle5` to the conda dependencies (instead of `pip` section) and compilers should be pulled automatically @andrewfullard.

Remeber the reason of `pickle5` is explained here https://github.com/tardis-sn/tardis/pull/1535 and would be solved once we get rid of `pyne` (or `pyne` gets updated on macOS).

**Comment by epassaro** [comment]
Closed by #1566
